app/src/main/java/com/amaze/filemanager/utils/ServiceWatcherUtil.java

```diff
@@ -25,26 +30,27 @@
     private Handler handler;
     private static HandlerThread handlerThread;
     private ProgressHandler progressHandler;
-    long totalSize;
     private Runnable runnable;
 
+    public static int STATE = -1;
+
     private static ArrayList<Intent> pendingIntents = new ArrayList<>();
 
     // position of byte in total byte size to be copied
-    public static long POSITION = 0L;
+    public static volatile long POSITION = 0L;
```

Comment:
Naming conventions, this variable in not final. Also, volatile is a strong word, are you sure there isn't another way? (Maybe 'atomic' is what you want).

Reply:
Yes, volatile fits the usage here.

Reply:
Why not atomic?

Reply:
Methods are atomic, not variables. Variables are volatile. As for using wrapper classes of Atomic behaviour, I don't see the need.